kartothek/io_components/metapartition.py

```diff
@@ -852,7 +852,11 @@ def _reconstruct_index_columns(
         if cleaned_original_columns != original_columns:
             # indexer call is slow, so only do that if really necessary
             df = df.loc[:, cleaned_original_columns]
-        return pd.concat(index_cols + [df], axis=1, sort=False, join="inner")
+
+        if len(index_cols) > 0:
+            df = pd.concat(index_cols + [df], axis=1, sort=False, join="inner")
```

Comment:
I'm ok with the change but do we understand what's causing this? Why is concat slow if the input is a single item? That should be a no-op. Is this worth reporting to pandas?